apps/openmw/mwscript/statsextensions.cpp

```diff
@@ -1019,7 +1019,19 @@ namespace MWScript
                     if (ptr == player)
                         return;
 
-                    ptr.getClass().getNpcStats(ptr).raiseRank(factionID);
+                    // If we already changed rank for this NPC, modify current rank in the NPC stats.
+                    // Otherwise take rank from base NPC record, increase it and put it to NPC data.
+                    int currentRank = ptr.getClass().getNpcStats(ptr).getFactionRank(Misc::StringUtils::lowerCase(factionID));
```

Comment:
NPC stats functions including the newly added one will normalize the faction ID automatically and other calls of them here and in LowerRank make the assumption that it will always happen so this lowerCase operation and the same operation in LowerRank is kind of redundant.